graphql/tests/mod_test.ts

```diff
@@ -0,0 +1,101 @@
+import { Rhum } from "../../test_deps.ts";
+import { GraphQL } from "../mod.ts";
+import { buildSchema } from "./deps.ts";
+import { Drash } from "../../deps.ts";
+
+//
+// DATA
+//
+const schema = buildSchema(`
+  type Query {
+    hello: String
+  }
+`);
+const root = {
+  hello: () => {
+    return "Hello world!";
+  },
+};
+const graphQL = GraphQL({ schema, graphiql: true, rootValue: root });
+class GraphQLResource extends Drash.Http.Resource {
+  static paths = ["/graphql"];
+
+  @Drash.Http.Middleware({
+    before_request: [graphQL],
+  })
+  // Used purely as an endpoint for the frontend playground
+  public GET() {
+    return this.response;
+  }
+
+  // Used purely as an endpoint to make a query from the client
+  @Drash.Http.Middleware({
+    before_request: [graphQL],
+    after_request: [],
+  })
```

Comment:
i still don't get why this is also in line 23? can we remove one of these

Reply:
also interesting pattern of having the middleware in the resource. so the middleware only invokes for this endpoint?

Reply:
yea it was breno's idea. so we have server level and more granular middleware (resource level)

Reply:
Yeah we should put it on class level, i think we'd just move it to be above the `class Res extends ...`, and remove those lines above the resource methods

Reply:
https://github.com/drashland/deno-drash-middleware/pull/118/commits/1505f8d4f5aeee176aec09c35709217a525ae543